### ISSUE TITLE
feat: send transactional email to organization owners/admins when a new member joins the org

### ISSUE THREAD

**Issue by marcklingen** [langfuse/langfuse#3504]
See issue title, Langfuse already has logic based on react.email and nodemailer for transactional emails (invites, password reset). From a security perspective, org admins/owners should be notified whenever a new person joins the organization.

**Comment by memyselfandglitch** [comment]
Hi I would like to work on this. It is my first time contributing to Langfuse.

**Comment by mohanadkandil** [comment]
Hello, I'd love to contribute to this issue. 
The approach is a simple -> New member notification template which triggered when a new member is added.

**Comment by memyselfandglitch** [comment]
Hi @jannikmaierhoefer I just have a few questions while going through the codebase. While a user is not created but invited by the owner, the mutCreateProjectOwner route seems to be triggered in components which is created in api.members.create.useMutation in the file CreateTRPCReact.tsx. As far I understood the user doesnt have the option to accept the invite they just are a part, if invited. My doubt is that if the user is already registered, he doesn't need to accept the invite so the mail doesn't need to be sent, but if he hasn't registered then when he registers, the mail is to be sent to the owners. I haven't worked with TRPC so I just need a bit of guidance if possible.

**Comment by marcklingen** [comment]
This is correct @memyselfandglitch, I think constraining this transactional email to either trigger when someone joined (signed up while having an invite which translates into being a member of the organization) or when someone was added (existing users can be added right away without an invite-accept process). Probably adjusting the message of the email a bit makes sense here to reflect this ("was added" vs "joined the organization"). What do you think?

**Comment by mohanadkandil** [comment]
Hi @marcklingen ,

I see that someone is already actively working on this issue, so I’ll close my pull request for now. However, @memyselfandglitch, feel free to take some inspiration from the code I’ve submitted if it’s helpful in your implementation!

Looking forward to seeing the feature completed. Thanks.

**Comment by memyselfandglitch** [comment]
Hello @marcklingen, yes I agree it would be better to have a conditional mail sent to the owner/admins depending on whether the user already exists or signs up. Taking inspiration from @mohanadkandil 's PR maybe I could create a membershipNotificationEmailTemplate additionally which would create the controller for sendMembershipNotificationEmail which could take in a boolean parameter of the added user existing in the db or not. This would be used in the memberRouter which would be invoked in two regions, firstly where user exists but organization membership doesn't exist yet(sending the "joined organization" message) and secondly where user doesn't exist (sending the "was added" mail). Is my approach accurate?

**Comment by memyselfandglitch** [comment]
Hello @jannikmaierhoefer I have created a PR. I shall test it locally using NodeMailer and update in sometime. I have some confusion about the style guidelines using prettier. Also could you inform about the tests failing or is it just related to authorization..

**Comment by marcklingen** [comment]
thank you, I will review this soon! (today crazy busy, hope I'll get to it tomorrow)
